main.js

```diff
@@ -1,98 +1,141 @@
-  meditate.classList.add('buttonChange');
+  meditateButton.classList.add('btn-category');
 };
 
 function exercise() {
   category = "Exercise";
-  exercise.classList.add('buttonChange');
-};
-
-beginTimer() {
-  currentActivity.startTimer;
+  exerciseButton.classList.add('btn-category');
 };
 
 // HELPER FUNCTIONS
 
-
-var startMinutes = minutes.value;// input
-var startSeconds = seconds.value;
-var second = 1000;
-var minute = second * 60;
-var timer;
-var minutesLeft = document.querySelector('.minutesLeft');
-var secondsLeft = document.querySelector('.secondsLeft');
-
-
 function createActivity() {
   var userActivity = {};
   userActivity.category = category;
-  userActivity.description = description.value;
-  userActivity.minutes = minutes.value;
-  userActivity.seconds = seconds.value;
-  return userActivity;0
+  if (description.value != "") {
+    userActivity.description = description.value;
+  } //else showError('desc');
+  if (verifyNumber(inputMinutes.value)) {
+    userActivity.minutes = inputMinutes.value;
+  } //else showError('min');
+  if (verifyNumber(inputSeconds.value)) {
+    userActivity.seconds = inputSeconds.value;
+  } //else showError('sec');
+  return userActivity;  // to instantiation below
+};
+
+var descriptionError = document.querySelector('.description-error');
+var minutesError = document.querySelector('.minutes-error');
+var secondsError = document.querySelector('.seconds-error');
+
+function showError(data) {
+  if (data = 'desc') {
+    show(descriptionError);
+  };
+  if (data = 'min') {
+    show(minutesError);   //document minute error message
+  };
+  if (data = 'sec') {
+    show(secondsError)
+  };
+};
+
+function verifyNumber(data) {
+  if (isNaN(data) || data < 0 || data > 300) {
+    return false;
+  } else return true;
 };
 
-// start Activity button
+// start Activity button = store activity
 function storeActivity() {
   hideForm();
   currentActivity = new Activity(createActivity());
+  console.log(currentActivity);
   clearForm();
+  setTimer();
+};
+
+
+function setTimer() {
+  minutes = currentActivity.minutes;
+  seconds = currentActivity.seconds;
+  updateTimer();
+};
+
+function updateTimer() {
+  minutesLeft.innerHTML = minutes;
+  if (seconds < 10) {
+    secondsLeft.innerHTML = "0" + seconds; // keeps seconds inline
+  } else secondsLeft.innerHTML = seconds;
+};
+
+function beginTimer() {
+  currentActivity.startTimer();
 };
 
 function showRemaining() {
-  userDescription.innerText = userActivity['description']; // show second page descriptor
-  minutesLeft.innerHTML = minutes - 1;
-  secondsLeft.innerHTML = seconds - 1;
-  if (minutes === 0 && seconds === 0) {
-    clearInterval(timer);
+  //userDescription.innerText = userActivity['description']; // show second page descriptor
+  if (seconds === -1) {
+    minutes--;
+    seconds = 59;
+  };
+  if (minutes == 0 && seconds == 0) {
+    updateTimer(); //clearInterval(timer);
+    timerDialogue.innerText = "Complete"
```

Comment:
Looking forward to testing this timer complete when we get further into the project